systemtest/src/test/java/io/strimzi/systemtest/operators/topic/TopicReplicasChangeST.java

```diff
@@ -0,0 +1,497 @@
+@Tag(CRUISE_CONTROL)
+@BTONotSupported
+/**
+ * This class contains system tests for KafkaTopic resource changes, specifically focusing on testing replication factor adjustments
+ * under various conditions. It simulates real-world scenarios such as insufficient brokers for the desired replication factor,
+ * positive and negative replication factor changes, and recovery from failure states in the Kafka cluster or related components.
+ *
+ * The tests are structured to validate:
+ * - The ability to handle KafkaTopic creation with a replication factor higher than available brokers, including error handling and retries.
+ * - Positive testing scenarios where replication factors are increased and decreased successfully, ensuring the system correctly applies these changes.
+ * - Negative testing scenarios where replication factor changes to a value higher than available brokers are attempted, verifying the system's response and error handling.
+ * - Recovery of replication factor changes during crashes or restarts of critical components like Cruise Control and the Entity Operator, ensuring the system's resilience and fault tolerance.
+ *
+ * Each test case follows a specific scenario, with steps to manipulate the KafkaTopic's replication factor and validate the system's response and state transitions.
+ * These tests ensure the robustness of KafkaTopic management and the system's ability to handle configuration changes and recover from errors.
+ */
+
+public class TopicReplicasChangeST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(TopicST.class);
+
+    private TestStorage sharedTestStorage;
+    private String scraperPodName;
+
+    /**
+     * @description Verifies the system's handling of a KafkaTopic with a replication factor higher than the number of available brokers.
+     * It tests the system's error recognition and handling capabilities, the ability to adjust to a valid replication factor,
+     * and the successful creation and operational status of the topic.
+     *
+     * @steps
+     *  1. - Create a KafkaTopic with more replicas than available brokers to induce a configuration error.
+     *     - The topic exists in Kubernetes but not in Kafka, indicating a configuration error.
+     *  2. - Check for specific error messages indicating the cause of the failure.
+     *     - Specific error messages related to the replication factor issue are present.
+     *  3. - Correct the issue by setting a valid replication factor and ensure changes are applied successfully.
+     *     - Topic's replication factor is adjusted, and the topic becomes operational.
+     *
+     * @usecase
+     *  - Replication factor management
+     *  - Error handling and recovery
+     */
+    @ParallelTest
+    void testMoreReplicasThanAvailableBrokersWithFreshKafkaTopic() {
+        final TestStorage testStorage = new TestStorage(ResourceManager.getTestContext());
+        final int topicPartitions = 5;
```

Comment:
does topicPartition play more important role than once set when creating topic ?

Reply:
I don't think so it plays a more important role in this case.